lab-judy/test/cowsay-controller-test.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+require('./lib/test-setup');
+
+const cowsay = require('cowsay-browser');
+const angular = require('angular');
+
+describe('testing cowsayCtrl', function(){
+  beforeEach(() => {
+    angular.mock.module('demoApp');
+    angular.mock.inject($controller => {
+      this.cowsayCtrl = new $controller('CowsayController');
+    });
+  });
+
+
+  describe('testing initial properties', () => {
+
+    //1. Test for title = Mooo
+    it('should have title = Moooo', () => {
+      expect(this.cowsayCtrl.title).toBe('Moooo');
+    });
+
+    //2. Test for empty history array
+    it('should have empty history array', () => {
+      expect(Array.isArray(this.cowsayCtrl.history)).toBe(true);
+    });
+
+    //3. Test that cowfiles = cowsay.list
+    it('should have cowfiles = cowsay.list', () => {
+      cowsay.list((err, list) => {
+        expect(this.cowsayCtrl.cowfiles).toEqual(list);
+        expect(this.cowsayCtrl.currentCow).toEqual(list[0]);
+      });
+    });
+  });
+
+  describe('testing #updateCow()', () => {
+
+    //4. Test that #updateCow returns Beavis
+    it('should return beavis with hello', () => {
+      let expectedResult = '\n' + cowsay.say({text: 'hello', f: this.cowsayCtrl.currentCow});
+      let result = this.cowsayCtrl.updateCow('hello');
+      expect(result).toEqual(expectedResult);
+    });
+  });
+
+  describe('testing #speak()', () => {
+    //5. Test that #speak returns Beavis
+    it('should return beavis with hello', () => {
+      let expectedResult = '\n' + cowsay.say({text: 'hello', f: this.cowsayCtrl.currentCow});
+      this.cowsayCtrl.speak('hello');
+
+      expect(this.cowsayCtrl.spoken).toEqual(expectedResult);
+      expect(this.cowsayCtrl.history[0]).toEqual(expectedResult);
+    });
+  });
+
+  describe('testing #undo()', () => {
+
+    //6. Test that #undo works properly with 2 inputs
+    it('should return empty array', () => {
+      let expectedResult = '\n' + cowsay.say({text: 'hello', f: this.cowsayCtrl.currentCow});
+      this.cowsayCtrl.speak('hello');
+      this.cowsayCtrl.speak('hello again');
+      this.cowsayCtrl.undo();
+      expect(this.cowsayCtrl.spoken).toEqual(expectedResult);
+      expect(this.cowsayCtrl.history.length).toEqual(0);
+    });
+
+    //7. Test that #undo works properly with 3 inputs
```

Comment:
Instead of testing that your function works properly with multiple numbers of inputs, I would do a reasonable number (like 3) in one test, and then test your default functionality of your two methods... for example, what should the cow be saying if there is no input or if someone got rid of all of the history by pressing undo many times?